### DISCUSSION TITLE
Fetching Meta Data (title and description etc.) from an api before page serve - SSR

### DISCUSSION THREAD

**Discussion by akshay-sood** [quasarframework/quasar#9520] [CLI - SSR mode] [answered]
Hi guys,

I am using quasar for building apps since November 2020, and I must say, it's amazing and super wonderful!

I now tried to give it a try to build an SSR website. And am stuck badly here:

I understand this might come off as a total noob question, but no matter what I do, for the life of me, I can not figure out how to fetch the metadata from a backend API and set it using createMetaMixin (Quasar v2). No matter whatever I do, the remote API is always fetched after the page has been rendered and sent to the client, so meta tags are set after page render.

I am using Options API.

Can someone please help me out with this?

**Comment by dmitry** [comment, marked as answer]
For the global API requests you may use a [boot files](https://quasar.dev/quasar-cli/boot-files), as they allow you to use async/await.
For the specific pages there is a special method named [preFetch](https://quasar.dev/quasar-cli/prefetch-feature) which is called everytime router uses the page in the backend.

**Reply by akashramtohul4** [reply]
i dont know if that can work for you i also have being using the SSR mode with the help of Axios .
am adding the link please check if that can help 

https://jasonwatmore.com/post/2020/07/23/vue-axios-http-post-request-examples

**Reply by akshay-sood** [reply]
Thank you for the reply @akashramtohul4 I was trying to do the same, but the problem seems to be, that Quasar createMetaMixin appears to run even before the created hook. (Unless I am completely off the mark here), so in created hook, it actually fetches the details after the meta details have been set.

**Reply by akshay-sood** [reply]
@dmitry Thank you for your prompt response! I also was looking into the preFetch hook. And I am going to try it now.

However, I found two things: 

1. I would have to use store to use data fetched in preFetch - there is no other way to pass it to the createMetaMixin
2. The preFetch hook doesn't work when I put it in a mixin (probably because the component has still not been created), so I have to repeat the same preFetch hook and call to fetch and store code again and again in all the pages. 

Any workarounds for these please?

**Reply by akshay-sood** [reply]
@dmitry Thank you so much for pointing me in the right direction. I have almost resolved the issue!

Just two things though:

1. Any idea, how to handle preFetch when the same component is rendered on different routes? e.g: the Service Detail page which can have different services based on the route? I  think preFetch is not re-running in this scenario. However, going from 1 component to another works fine, and also ssr on all pages.

2. Is there any possibility to use preFetch in a mixin? It would save me a bit of redundant code.

**Reply by KarelDiaz** [reply]
Im having the same problem

**Reply by KarelDiaz** [reply]
@akshay-sood could you solve that?

**Reply by akshay-sood** [reply]
@KarelDiaz which problem? Having prefetch code in mixing? I don't remember what I ended up doing there, but if you can tell me exact issue you are facing, I can try to find out what I did about that.

**Reply by KarelDiaz** [reply]
@akshay-sood 
Fetching Meta Data (title and description etc.) from an api before page serve - SSR
